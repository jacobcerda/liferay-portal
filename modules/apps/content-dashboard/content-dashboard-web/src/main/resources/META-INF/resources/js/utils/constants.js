/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

export const BAR_CHART = {
	axisMargin: 8,
	barHeight: 16,
	dotRadiusMax: 35,
	dotRadiusMin: 5,
	emptyHeight: 300,
	fillXAxis: '#666',
	height: 400,
	legendHeight: 70,
	stroke: '#E7E7ED',
	width: 1150,
};

export const COLORS = [
	'#4B9BFF',
	'#50D2A0',
	'#FFB46E',
	'#b19cd9',
	'#FFC0CB',
	'#AFEEEE',
	'#009975',
	'#9B111E',
	'#FFDF00',
];
