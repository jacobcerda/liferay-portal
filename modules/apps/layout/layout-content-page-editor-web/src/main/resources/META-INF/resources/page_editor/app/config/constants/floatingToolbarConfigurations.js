/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {CollectionConfigurationPanel} from '../../../app/components/floating-toolbar/CollectionConfigurationPanel';
import {FragmentConfigurationPanel} from '../../../app/components/floating-toolbar/FragmentConfigurationPanel';
import {ImagePropertiesPanel} from '../../../app/components/floating-toolbar/ImagePropertiesPanel';
import {MappingPanel} from '../../../app/components/floating-toolbar/MappingPanel';
import {RowConfigurationPanel} from '../../../app/components/floating-toolbar/RowConfigurationPanel';
import {ContainerConfigurationPanel} from '../../components/floating-toolbar/ContainerConfigurationPanel';
import ContainerLinkPanel from '../../components/floating-toolbar/ContainerLinkPanel';
import EditableLinkPanel from '../../components/floating-toolbar/EditableLinkPanel';
import {SectionConfigurationPanel} from '../../components/floating-toolbar/SectionConfigurationPanel';
import {config} from '../index';
import {EDITABLE_FLOATING_TOOLBAR_BUTTONS} from './editableFloatingToolbarButtons';
import {LAYOUT_DATA_FLOATING_TOOLBAR_BUTTONS} from './layoutDataFloatingToolbarButtons';

const {
	collectionConfiguration,
	containerConfiguration,
	containerLink,
	fragmentConfiguration,
	rowConfiguration,
} = LAYOUT_DATA_FLOATING_TOOLBAR_BUTTONS;

const {imageProperties, link, map} = EDITABLE_FLOATING_TOOLBAR_BUTTONS;

export const FLOATING_TOOLBAR_CONFIGURATIONS = {
	[collectionConfiguration.panelId]: CollectionConfigurationPanel,
	[containerConfiguration.panelId]: config.containerItemEnabled
		? ContainerConfigurationPanel
		: SectionConfigurationPanel,
	[containerLink.panelId]: ContainerLinkPanel,
	[fragmentConfiguration.panelId]: FragmentConfigurationPanel,
	[imageProperties.panelId]: ImagePropertiesPanel,
	[link.panelId]: EditableLinkPanel,
	[map.panelId]: MappingPanel,
	[rowConfiguration.panelId]: RowConfigurationPanel,
};
