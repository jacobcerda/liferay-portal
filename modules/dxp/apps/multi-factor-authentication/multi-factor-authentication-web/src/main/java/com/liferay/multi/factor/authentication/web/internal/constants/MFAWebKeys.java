/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * The contents of this file are subject to the terms of the Liferay Enterprise
 * Subscription License ("License"). You may not use this file except in
 * compliance with the License. You can obtain a copy of the License by
 * contacting Liferay, Inc. See the License for the specific language governing
 * permissions and limitations under the License, including but not limited to
 * distribution rights of the Software.
 *
 *
 *
 */

package com.liferay.multi.factor.authentication.web.internal.constants;

/**
 * @author Marta Medio
 */
public class MFAWebKeys {

	public static final String MFA_BROWSER_CHECKER = "MFA_BROWSER_CHECKER";

	public static final String MFA_BROWSER_CHECKER_NAME =
		"MFA_BROWSER_CHECKER_NAME";

	public static final String MFA_BROWSER_CHECKERS = "MFA_BROWSER_CHECKERS";

	public static final String MFA_USER_ID = "MFA_USER_ID";

	public static final String MFA_WEB_DIGEST = "MFA_WEB_DIGEST";

	public static final String MFA_WEB_KEY = "MFA_WEB_KEY";

}