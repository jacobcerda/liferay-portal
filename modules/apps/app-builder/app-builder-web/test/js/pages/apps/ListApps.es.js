/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {waitForElementToBeRemoved} from '@testing-library/dom';
import {cleanup, fireEvent, render} from '@testing-library/react';
import React from 'react';

import ListApps from '../../../../src/main/resources/META-INF/resources/js/pages/apps/ListApps.es';
import * as time from '../../../../src/main/resources/META-INF/resources/js/utils/time.es';
import AppContextProviderWrapper from '../../AppContextProviderWrapper.es';
import {RESPONSES} from '../../constants.es';

const DRODOWN_VALUES = {
	items: [
		{
			id: 37568,
			name: {
				'en-US': 'test',
			},
		},
	],
};

describe('ListApps', () => {
	beforeEach(() => {
		jest.spyOn(time, 'fromNow').mockImplementation(() => 'months ago');
	});

	afterEach(() => {
		cleanup();
		jest.restoreAllMocks();
	});

	it('renders', async () => {
		fetch.mockResponseOnce(JSON.stringify(RESPONSES.ONE_ITEM));
		fetch.mockResponse(JSON.stringify(DRODOWN_VALUES));

		const {asFragment} = render(
			<ListApps match={{params: {dataDefinitionId: ''}, url: '/'}} />,
			{wrapper: AppContextProviderWrapper}
		);

		await waitForElementToBeRemoved(() =>
			document.querySelector('span.loading-animation')
		);

		expect(asFragment()).toMatchSnapshot();
	});

	it('renders with dataDefinitionId and 5 apps in the list', async () => {
		fetch.mockResponseOnce(JSON.stringify(RESPONSES.MANY_ITEMS(5)));
		fetch.mockResponse(JSON.stringify(DRODOWN_VALUES));

		const {container} = render(
			<ListApps match={{params: {dataDefinitionId: '1'}, url: '/'}} />,
			{wrapper: AppContextProviderWrapper}
		);

		await waitForElementToBeRemoved(() =>
			document.querySelector('span.loading-animation')
		);

		expect(container.querySelector('tbody').children.length).toEqual(5);
	});

	it('renders with no dataDefinitionId, opens a new app popover and lists 5 apps', async () => {
		fetch.mockResponseOnce(JSON.stringify(RESPONSES.MANY_ITEMS(5)));
		fetch.mockResponse(JSON.stringify(DRODOWN_VALUES));

		const {container} = render(
			<ListApps match={{params: {dataDefinitionId: ''}, url: '/'}} />,
			{wrapper: AppContextProviderWrapper}
		);

		await waitForElementToBeRemoved(() =>
			document.querySelector('span.loading-animation')
		);

		expect(container.querySelector('tbody').children.length).toEqual(5);

		const newAppButton = document.querySelector(
			'.nav-btn.nav-btn-monospaced.btn.btn-monospaced.btn-primary'
		);

		fireEvent.click(newAppButton);

		expect(
			document.querySelector('.popover.apps-popover.mw-100')
		).toBeTruthy();

		fireEvent.click(newAppButton);

		expect(document.querySelector('.popover.apps-popover.mw-100.hide'));

		await fireEvent.click(newAppButton);

		await fireEvent.click(
			document.querySelector('.d-flex.justify-content-end').children[0]
		);

		expect(document.querySelector('.popover.apps-popover.mw-100.hide'));
	});

	it('renders with empty state', async () => {
		fetch.mockResponseOnce(JSON.stringify(RESPONSES.NO_ITEMS));
		fetch.mockResponse(JSON.stringify(DRODOWN_VALUES));

		const {container} = render(
			<ListApps match={{params: {dataDefinitionId: ''}, url: '/'}} />,
			{wrapper: AppContextProviderWrapper}
		);

		await waitForElementToBeRemoved(() =>
			document.querySelector('span.loading-animation')
		);

		expect(
			container.querySelector('.taglib-empty-result-message-title')
				.textContent
		).toEqual('there-are-no-apps-yet');
	});
});
