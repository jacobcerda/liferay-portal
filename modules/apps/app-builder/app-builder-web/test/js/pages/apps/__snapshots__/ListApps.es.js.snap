// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListApps renders 1`] = `
<DocumentFragment>
  <div
    class="tools-control-group"
  >
    <div
      class="control-menu-level-1-heading"
    />
  </div>
  <nav
    class="management-bar navbar navbar-expand-md management-bar-light"
  >
    <div
      class="container-fluid container-fluid-max-xl"
    >
      <ul
        class="navbar-nav"
      >
        <li
          class="nav-item"
        >
          <div
            class="dropdown"
          >
            <button
              class="dropdown-toggle nav-link btn btn-unstyled"
              id="filter-and-order"
              style=""
              type="button"
            >
              <span
                class="navbar-breakpoint-down-d-none"
              >
                filter-and-order
                <svg
                  class="lexicon-icon lexicon-icon-caret-bottom inline-item inline-item-after"
                  role="presentation"
                >
                  <use
                    xlink:href="#caret-bottom"
                  />
                </svg>
              </span>
              <span
                class="navbar-breakpoint-d-none"
              >
                <svg
                  class="lexicon-icon lexicon-icon-filter inline-item inline-item-after"
                  role="presentation"
                >
                  <use
                    xlink:href="#filter"
                  />
                </svg>
              </span>
            </button>
          </div>
        </li>
        <li
          class="nav-item"
        >
          <button
            class="nav-link nav-link-monospaced order-arrow-down-active btn btn-monospaced btn-unstyled"
            data-title="reverse-sort-direction"
            type="button"
          >
            <svg
              class="lexicon-icon lexicon-icon-order-arrow"
              role="presentation"
            >
              <use
                xlink:href="#order-arrow"
              />
            </svg>
          </button>
        </li>
      </ul>
      <div
        class="navbar-form navbar-form-autofit navbar-overlay navbar-overlay-sm-down"
      >
        <div
          class="container-fluid container-fluid-max-xl"
        >
          <form
            role="search"
          >
            <div
              class="input-group"
            >
              <div
                class="input-group-item"
              >
                <input
                  aria-label="search"
                  class="form-control input-group-inset input-group-inset-after"
                  placeholder="search..."
                  type="text"
                  value=""
                />
                <span
                  class="input-group-inset-item input-group-inset-item-after"
                >
                  <button
                    class="navbar-breakpoint-d-none btn btn-monospaced btn-unstyled"
                    type="button"
                  >
                    <svg
                      class="lexicon-icon lexicon-icon-times"
                      role="presentation"
                    >
                      <use
                        xlink:href="#times"
                      />
                    </svg>
                  </button>
                  <button
                    class="btn btn-monospaced btn-unstyled"
                    type="submit"
                  >
                    <svg
                      class="lexicon-icon lexicon-icon-search"
                      role="presentation"
                    >
                      <use
                        xlink:href="#search"
                      />
                    </svg>
                  </button>
                </span>
              </div>
            </div>
          </form>
        </div>
      </div>
      <ul
        class="navbar-nav"
      >
        <li
          class="nav-item navbar-breakpoint-d-none"
        >
          <button
            class="nav-link nav-link-monospaced btn btn-unstyled"
            type="button"
          >
            <svg
              class="lexicon-icon lexicon-icon-search"
              role="presentation"
            >
              <use
                xlink:href="#search"
              />
            </svg>
          </button>
        </li>
        <li
          class="nav-item"
        >
          <button
            class="nav-btn nav-btn-monospaced btn btn-monospaced btn-primary"
            data-title="new-app"
            type="button"
          >
            <svg
              class="lexicon-icon lexicon-icon-plus"
              role="presentation"
            >
              <use
                xlink:href="#plus"
              />
            </svg>
          </button>
        </li>
      </ul>
    </div>
  </nav>
  <div
    class="container-fluid container-fluid-max-xl"
  >
    <div>
      <div
        class="table-responsive"
      >
        <table
          class="table table-autofit table-list"
        >
          <thead>
            <tr
              class=""
            >
              <th
                class="table-cell-expand text-left"
              >
                name
              </th>
              <th
                class="table-cell-expand-smaller text-left"
              >
                object
              </th>
              <th
                class="table-cell-expand-smaller text-left"
              >
                deployed-as
              </th>
              <th
                class="table-cell-expand-smaller text-left"
              >
                create-date
              </th>
              <th
                class="table-cell-expand-smaller text-left"
              >
                modified-date
              </th>
              <th
                class="table-cell-expand-smaller text-left"
              >
                status
              </th>
              <th
                class=""
              />
            </tr>
          </thead>
          <tbody>
            <tr
              class=""
            >
              <td
                class="table-cell-expand text-left"
              >
                <p
                  class="table-list-title"
                >
                  Item 1
                </p>
              </td>
              <td
                class="table-cell-expand-smaller text-left"
              >
                Object
              </td>
              <td
                class="table-cell-expand-smaller text-left"
              >
                standalone
              </td>
              <td
                class="table-cell-expand-smaller text-left"
              >
                months ago
              </td>
              <td
                class="table-cell-expand-smaller text-left"
              >
                months ago
              </td>
              <td
                class="table-cell-expand-smaller text-left"
              >
                <span
                  class="label label-success"
                >
                  <span
                    class="label-item label-item-expand"
                  >
                    deployed
                  </span>
                </span>
              </td>
              <td
                class=""
              >
                <div
                  class="dropdown dropdown-action"
                >
                  <button
                    class="dropdown-toggle page-link btn btn-unstyled"
                    style=""
                    type="button"
                  >
                    <svg
                      class="lexicon-icon lexicon-icon-ellipsis-v"
                      role="presentation"
                    >
                      <use
                        xlink:href="#ellipsis-v"
                      />
                    </svg>
                  </button>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    class="popover apps-popover mw-100 hide"
  >
    <div
      class="arrow"
    />
    <div
      class="border-0 popover-header"
    >
      <h4
        class="mb-3"
      >
        new-app
      </h4>
      <span
        class="font-weight-light text-secondary"
      >
        create-an-app-to-manage-the-data-of-an-object
      </span>
    </div>
    <div
      class="popover-body"
    >
      <label>
        object
      </label>
      <div
        class="dropdown"
        label="select-object"
      >
        <button
          class="dropdown-toggle clearfix w-100 btn btn-secondary"
          style=""
          type="button"
        >
          <span
            class="float-left"
          >
            select-object
          </span>
          <svg
            class="lexicon-icon lexicon-icon-caret-bottom float-right icon"
            role="presentation"
          >
            <use
              xlink:href="#caret-bottom"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="popover-footer"
    >
      <div
        class="border-top mt-3 p-3"
        style="width: 450px;"
      >
        <div
          class="d-flex justify-content-end"
        >
          <button
            class="mr-3 btn btn-sm btn-secondary"
            type="button"
          >
            cancel
          </button>
          <button
            class="btn btn-sm btn-primary"
            disabled=""
            type="button"
          >
            continue
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
